### PR TITLE
remove opaque parameters from delete request and return bucket id o…

### PR DESCRIPTION
**Pull request by wlan0**
…n succesful bucket creation.

### CODE REVIEW

File: cosi.proto [hunk 1]
```diff
@@ -182,19 +182,15 @@ message ProvisionerCreateBucketRequest {
 }
 
 message ProvisionerCreateBucketResponse {
-    // Intentionally left blank
+    // BucketName returned here is expected to be the globally unique 
+	// identifier for the bucket in the object storage provider 
+	string BucketName = 1;
```
Comment (jeffvance) [inline cosi.proto:187]: do we want BucketName here or BucketID?
I thought the response was going to return the full bucket it.

File: cosi.proto [hunk 2]
```diff
@@ -182,19 +182,15 @@ message ProvisionerCreateBucketRequest {
...
 }
 
 message ProvisionerDeleteBucketRequest {
     // This field is REQUIRED
     // Protocol specific information required by the call is passed in as key,value pairs.
     Protocol protocol = 1;
-
```
Comment (jeffvance) [inline cosi.proto:192]: don't we need the bucket id? How does the provisoner know which bucket to delete?

File: cosi.proto [hunk 3]
```diff
@@ -182,19 +182,15 @@ message ProvisionerCreateBucketRequest {
...
-    // This field is OPTIONAL
-    // Protocol specific information required by the call is passed in as key,value pairs.
-    // The caller should treat the values in parameters as opaque. 
-    // The receiver is responsible for parsing and validating the values.
-    map<string,string> parameters = 2;
```
Comment (jeffvance) [inline cosi.proto:197]: In ProvisionerDeleteBucketResponse (L200) should access_policy = 1 (not 4)?

Comment (jeffvance) [reply cosi.proto:197]: In ProvisionerGrantBucketAccessRequest (L204), I thought that protocol and parameters were being removed

File: cosi.proto [hunk 4]
```diff
@@ -239,13 +235,9 @@ message ProvisionerRevokeBucketAccessRequest {
     // Protocol specific information required by the call is passed in as key,value pairs.
     Protocol protocol = 1;
 
-    // This field is OPTIONAL
```
[the remark this answers is not in the document]
Comment (wlan0) [reply cosi.proto:242]: Yes, that's correct. I've updated the PR

**Comment by jeffvance** [comment]
lgtm

File: cosi.proto [hunk 5]
```diff
@@ -203,27 +195,20 @@ message ProvisionerDeleteBucketResponse {
...
+    string access_policy = 2;
 }
 
 message ProvisionerGrantBucketAccessResponse {
     // This field is OPTIONAL
-    // This is the account that is being provided access. This will
+    // This is the accountID that is being provided access. This will
     // be required later to revoke access. 
-    string principal = 1;
+    string accountID = 1;
```
Comment (wlan0) [inline cosi.proto:211]: accountID seems more intuitive and also is symmetric with bucketID

File: spec.md [hunk 6]
```diff
@@ -266,15 +266,10 @@ message GCSParameters {
...
-    // version is the name of the protocol version
-    string version = 2;
-
     oneof type {
-        S3Parameters s3 = 3;
-        AzureBlobParameters azureBlob = 4;
-        GCSParameters gcs = 5;
+        S3Parameters s3 = 1;
+        AzureBlobParameters azureBlob = 2;
+        GCSParameters gcs = 3;
```
Comment (saad-ali) [inline spec.md:272]: nit: why suffix these with `Parameters`?

**Review by saad-ali — COMMENTED** [review]
LGTM mod `accountName` (I wasn't involved in the discussion to replace `principal` with `accountName`, so I'll assume those were discussed with the wider community).

File: spec.md (hunk 6 above)
Comment (wlan0) [reply spec.md:272]: Protocol name S3 otherwise collides with the Parameter type S3

**Comment by wlan0** [comment]
> LGTM mod accountName (I wasn't involved in the discussion to replace principal with accountName, so I'll assume those were discussed with the wider community).

Yeah, this was discussed with the community and accepted

File: spec.md (hunk 6 above)
Comment (saad-ali) [reply spec.md:272]: Do you mean it collides with `enum ProtocolName`? if so, is that enum still needed (it is removed in this PR)?

File: cosi.proto (hunk 5 above)
Comment (saad-ali) [reply cosi.proto:222]: Ack. I don't have the context on this, maybe worth having someone else who was part of that discussion take a look

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Pull request merged**